internal/management/controller/instance_startup.go

```diff
@@ -262,13 +263,7 @@ func (r *InstanceReconciler) verifyPgDataCoherenceForPrimary(ctx context.Context
 			return fmt.Errorf("while ensuring all WAL files are archived: %w", err)
 		}
 
-		// pg_rewind could require a clean shutdown of the old primary to
-		// work. Unfortunately, if the old primary is already clean starting
-		// it up may make it advance in respect to the new one.
-		// The only way to check if we really need to start it up before
-		// invoking pg_rewind is to try using pg_rewind and, on failures,
-		// retrying after having started up the instance.
-		err = r.instance.Rewind(ctx, pgVersion)
```

Comment:
I removed the comment because it is outdated. Since version 13 pg_rewind automatically makes sure that the instance is cleanly shutdown.